### DISCUSSION TITLE
Using the Connector as a proxy during Kafka communication

### DISCUSSION THREAD

**Discussion by boxer-code** [eclipse-edc/Connector#4433] [Q&A]
Hello everyone, I'm trying to implement a data exchange between two connectors and two kafka server. Is there a way to use the connector as you called it in a 2021 discussion, the "proxy mode"? It would be perfect if I could run the kafka server locally and the provider connector doesn't need to reach the consumer kafka. I'm aware that this creates a keyhole problem, but I don't want to exchange large amounts of data. It would be more important to be able to run the server locally and to be able to maintain only one url that needs to be reachable from the outside. Thanks in advance!

**Comment by ndr-brt** [comment]
My suggestion would be to take a look at the streaming samples: https://github.com/eclipse-edc/Samples/tree/main/transfer/streaming
you'll find some interesting examples

**Comment by boxer-code** [comment]
I already executed the samples and took a look at it. It has exactly the setup that I described above. The provider connector needs the possibility to access the kafka from the consumer. That's why I asked here if there is a possibility to skip this and only use the external endpoint from the connector.

**Reply by ndr-brt** [reply]
is the other way around, is the consumer that needs to be able to access to the kafka provider.
anyway the sample number 2, `kafka-to-http` permits to avoid that

**Reply by boxer-code** [reply]
Thanks for pointing to the example! But my understanding is that the http backend endpoint must be externally accessible?

**Reply by ndr-brt** [reply]
yes
